lib/smart_proxy_remote_execution_ssh/session.rb

```diff
@@ -30,7 +30,7 @@ def initialize_command
       end
       output_path = File.join(File.dirname(remote_script), 'output')
 
-      @connector.async_run("#{su_prefix}#{remote_script} | /usr/bin/tee #{output_path}") do |data|
+      @connector.async_run("set -o pipefail; #{su_prefix}#{remote_script} | /usr/bin/tee #{output_path}") do |data|
```

Comment:
This assumes the effective user's shell supports `pipefail`. Guess for now we should just mention it somewhere in the docs so the users won't be surprised by exit codes being wrong and/or messages like `set: Illegal option -o pipefail` in the output in case they're using something like `dash`.